src/js/environment/environment.ts

```diff
@@ -64,134 +72,130 @@ function supportsPassive(): boolean {
  * @property {boolean} safari - Is the browser Safari?
  * @property {EnvironmentVersion} version - The browser version.
  */
-export const Browser: GenericObject = {};
+export type BrowserEnvironment = {
+    androidNative: boolean;
+    chrome: boolean;
+    edge: boolean;
+    facebook: boolean;
+    firefox: boolean;
+    ie: boolean;
+    msie: boolean;
+    safari: boolean;
+    version: EnvironmentVersion;
+}
+export const Browser: BrowserEnvironment = {
+    get androidNative(): boolean {
+        return isAndroidNative();
+    },
+    get chrome(): boolean {
+        return isChrome();
+    },
+    get edge(): boolean {
+        return isEdge();
+    },
+    get facebook(): boolean {
+        return isFacebook();
+    },
+    get firefox(): boolean {
+        return isFF();
+    },
+    get ie(): boolean {
+        return isIE();
+    },
+    get msie(): boolean {
+        return isMSIE();
+    },
+    get safari(): boolean {
+        return isSafari();
+    },
+    get version(): EnvironmentVersion {
+        return browserVersion(this, userAgent);
```

Comment:
Is this cool? It used to be `memoize(browserVersion.bind(this, Browser, userAgent))`